docs/tutorial/tutorial-1.rst

```diff
@@ -0,0 +1,104 @@
+
+This tutorial assumes you've set up your environment as described in the
+:ref:`Getting started guide <get-started>`.
+
+Accessing NSURL
+---------------
+
+Start Python, and use ctypes to import a framework into the Python
+interpreter:
+
+.. code-block:: python
+
+    >>> from ctypes import cdll, util
+    >>> cdll.LoadLibrary(util.find_library("Foundation"))
+
+This loads Apple's Objective-C Foundation libraries into memory. You can now
+reference any class that is defined in that library. Next, we need to get a
+handle to the `NSURL` class:
+
+.. code-block:: python
+
+    >>> from rubicon.objc import ObjCClass
+    >>> NSURL = ObjCClass("NSURL")
+
+This gives us an `NSURL` class in Python which is transparently bridged to the
+`NSURL` class in the Objective-C runtime. Any method or property described in
+`Apple's documentation on NSURL <https://developer.apple.com/reference/foundation/nsurl?language=objc>`__  can be accessed over this bridge.
+
+Let's create an instance of an `NSURL` object. The `NSURL` documentation
+describes a static constructor `+URLWithString:`; we can invoke this
+constructor as::
+
+    >>> base = NSURL.URLWithString("http://pybee.org/")
+
+That is, the name of the method in Python is identical to the method in
+Objective-C. The first argument is declared as being an `NSString *`; Rubicon
+converts the Python `str` into an `NSString` instance as part of invoking the
+method.
+
+`NSURL` has another static constructor: `+URLWithString:relativeToURL:`. We
+can also invoke this constructor:
+
+    >>> full = NSURL.URLWithString("contributing/", relativeToURL=base)
+
+The second argument (`relativeToURL`) is accessed as a keyword argument. This
+argument is declared as being of type `NSURL *`; since `base` is an instance
+of NSURL, Rubicon can pass through this instance.
+
+Sometimes, an Objective-C method definition will use have the same keyword
+argument name twice. This is legal in Objective-C, but not in Python, as you
+can't repeat a keyword argument in a method call. In this case, you can use a
+"long form" of the method to explicitly invoke a descriptor by replacing
+colons with underscores::
+
+    >>> base = NSURL.URLWithString_("http://pybee.org/")
+    >>> full = NSURL.URLWithString_relativeToURL_("contributing", base)
+
+Instance methods
+----------------
+
+So far, we've been using the `+URLWithString:` static constructor. However, `NSURL`
+also provides an initializer method `-initWithString:`. To use this method, you
+first have to instruct the Objective-C runtime to allocate memory for the instance,
+then invoke the initializer:
+
+    >>> base = NSURL.alloc().initWithString("http://pybee.org/")
+
+Now that you have an instance of `NSURL`, you'll want to manipulate it.
+`NSURL` describes an `absoluteURL` property; this property can be
+accessed as a Python attribute::
+
+    >>> absolute = full.absoluteURL
+
+You can also invoke methods on the instance::
+
+    >>> full.toString()
```

Comment:
This should be `description`, there's no `toString` in Objective-C. (This might also be a good time to show that `str(...)` works on Objective-C objects.)

Reply:
ack.